cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -131,13 +132,14 @@ impl GrammarAST {
         }
     }
 
-    pub fn add_rule(&mut self, name: String, actiont: Option<String>) {
+    pub fn add_rule(&mut self, name: String, actiont: Option<String>, span: Option<Span>) {
```

Comment:
Something that just occurred to me: what is the span of? Is it the rule's name or the whole rule (including productions etc)?

Reply:
Ahh, this could be commented better, it is the LHS of the rule so it's name, the easiest place to see it is (I now see got cut off outside the diff context)

https://github.com/softdevteam/grmtools/pull/292/files/3f9278a8c9a06cacb99014bc7cfd3c0d8174314c#diff-f19cd37f82592a87f9e6c986053e2b5e5194c270f9ba8c6185ea5633125e0b74R1478-R1491

Reply:
OK, so if it's the name (which makes sense) then the signature should, I think, be:

```rust
 pub fn add_rule(&mut self, name: String, name_span: Option<Span>, actiont: Option<String>)
```

Reply:
Well, my link didn't seem to work, but it's line 1478-1491 of the function `test_token_spans` in grammar.rs